src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * #%L
+ * Alfresco Data model classes
+ * %%
+ * Copyright (C) 2005 - 2016 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software. 
+ * If the software was purchased under a paid Alfresco license, the terms of 
+ * the paid license agreement will prevail.  Otherwise, the software is 
+ * provided under the following open source license terms:
+ * 
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.service.cmr.repository;
+
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.alfresco.service.cmr.repository.datatype.DefaultTypeConverter;
+
+import junit.framework.TestCase;
+
+/**
+ * 
+ * @see org.alfresco.service.cmr.repository.Period
+ * @author Andreea Nechifor
+ *
+ */
+public class PeriodTest extends TestCase
+{
+
+    public void testMapToPeriod() throws ParseException
+    {
+        Map<String, String> periodArgs = new HashMap<String, String>();
+        periodArgs.put("periodType", "day");
+        Period period = new Period(periodArgs);
+        assertNotNull(period.getPeriodType());
+        assertNull(period.getExpression());
+        assertEquals(new Period(periodArgs), DefaultTypeConverter.INSTANCE.convert(Period.class, periodArgs));
```

Comment:
This class should just be testing the Period class (in particular the constructor that takes a `Map`). We don't need to test the `DefaultTypeConverter` here (and should not). This should be checking that after constructing a `Period` using a `Map`, that the state of the Period object is as we expect, e.g. `assertEquals("day", period.getPeriodType)`. The assertNotNull and assertNull are correctly checking the state of the Period.

Reply:
ok I modify .